awsagentprovider/src/test/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGeneratorTest.java

```diff
@@ -1029,6 +1030,84 @@ private void validateHttpStatusForNonLocalRootWithThrowableForClient(
     }
   }
 
+  @Test
+  public void testDBUserAttribute() {
+    mockAttribute(DB_SYSTEM, "db_system");
+    mockAttribute(DB_OPERATION, "db_operation");
+    mockAttribute(DB_STATEMENT, "db_statement");
+    mockAttribute(DB_USER, "db_user");
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    Attributes actualAttributes =
+        GENERATOR.generateMetricAttributeMapFromSpan(spanDataMock, resource).get(DEPENDENCY_METRIC);
+    assertThat(actualAttributes.get(AWS_REMOTE_OPERATION)).isEqualTo("db_operation");
+    assertThat(actualAttributes.get(AWS_REMOTE_DB_USER)).isEqualTo("db_user");
+  }
+
+  @Test
+  public void testDBUserAttributeAbsent() {
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    Attributes actualAttributes =
+        GENERATOR.generateMetricAttributeMapFromSpan(spanDataMock, resource).get(DEPENDENCY_METRIC);
+    assertThat(actualAttributes.get(AWS_REMOTE_DB_USER)).isNull();
+  }
+
+  @Test
+  public void testDBUserAttributeWithDifferentValues() {
+    mockAttribute(DB_SYSTEM, "db_system");
+    mockAttribute(DB_OPERATION, "db_operation");
+    mockAttribute(DB_STATEMENT, "db_statement");
+    mockAttribute(DB_USER, "non_db_user");
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    Attributes actualAttributes =
+        GENERATOR.generateMetricAttributeMapFromSpan(spanDataMock, resource).get(DEPENDENCY_METRIC);
+    assertThat(actualAttributes.get(AWS_REMOTE_DB_USER)).isEqualTo("non_db_user");
+  }
+
+  @Test
+  public void testDBUserAttributeNotPresentInServiceMetricForServerSpan() {
+    String dbUser = "db_user";
+    mockAttribute(DB_USER, dbUser);
+    when(spanDataMock.getKind()).thenReturn(SpanKind.SERVER);
+
+    Attributes actualAttributes =
+        GENERATOR.generateMetricAttributeMapFromSpan(spanDataMock, resource).get(SERVICE_METRIC);
+    assertThat(actualAttributes.get(AWS_REMOTE_DB_USER)).isNull();
+  }
+
+  @Test
+  public void testIsDbSpanTrueWhenDbSystemKeyIsPresent() {
+    mockAttribute(DB_SYSTEM, "DB system");
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    assertThat(AwsSpanProcessingUtil.isDBSpan(spanDataMock)).isTrue();
+  }
+
+  @Test
+  public void testIsDbSpanTrueWhenDbOperationKeyIsPresent() {
+    mockAttribute(DB_OPERATION, "DB operation");
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    assertThat(AwsSpanProcessingUtil.isDBSpan(spanDataMock)).isTrue();
+  }
+
+  @Test
+  public void testIsDbSpanFalseWhenOnlyDbUserKeyIsPresent() {
+    mockAttribute(DB_USER, "DB user");
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    assertThat(AwsSpanProcessingUtil.isDBSpan(spanDataMock)).isFalse();
+  }
+
+  @Test
+  public void testIsDbSpanFalse() {
+    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
+
+    assertThat(AwsSpanProcessingUtil.isDBSpan(spanDataMock)).isFalse();
+  }
```

Comment:
These tests are not really needed (we should already be testing `isDbSpan` logic in Util tests), but you are missing the one test you did need:
```suggestion
  @Test
  public void testDbUserPresentAndIsDbSpanFalse() {
    when(spanDataMock.getKind()).thenReturn(SpanKind.CLIENT);
    mockAttribute(DB_USER, "DB user");
    Attributes actualAttributes =
        GENERATOR.generateMetricAttributeMapFromSpan(spanDataMock, resource).get(SERVICE_METRIC);
    assertThat(actualAttributes.get(AWS_REMOTE_DB_USER)).isNull();
  }
```